### ISSUE TITLE
SC2039: False positives for type <thing>, esp. with dash, ash

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#788]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2039
- My shellcheck version (`shellcheck --version` or "online"): 0.3.7

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env ash

usage() {
    echo "Usage: $0 <name>"
    exit 0
}

if [ "$#" -ne 1 ]; then
    usage
fi

type "$1"
```

#### Here's what shellcheck currently says:

```
In lib/topical line 12:
type "$1"
^--  SC2039: In POSIX sh, 'type' is not supported.
```

#### Here's what I wanted or expected to see:

(no warning output)

**Comment by koalaman** [comment]
In 0.3.7, before `dash` support, `ash` was aliased to `sh`. Later this broke, so it defaulted to `bash`.

From fee6c94, `ash` will be aliased to `dash`, and there's a warning saying when this is happening.
